extensions/smallrye-reactive-messaging/deployment/src/main/java/io/quarkus/smallrye/reactivemessaging/deployment/SmallRyeReactiveMessagingProcessor.java

```diff
@@ -82,15 +84,27 @@ void validateBeanDeployment(
             }
         }
 
-        for (InjectionPointInfo injectionPoint : validationPhase.getContext().get(BuildExtension.Key.INJECTION_POINTS)) {
+        for (InjectionPointInfo injectionPoint : validationPhase.getContext()
+                .get(BuildExtension.Key.INJECTION_POINTS)) {
             if (injectionPoint.getRequiredType().name().equals(NAME_EMITTER)) {
                 AnnotationInstance stream = injectionPoint.getRequiredQualifier(NAME_STREAM);
                 if (stream != null) {
                     // Stream.value() is mandatory
                     String name = stream.value().asString();
+                    AnnotationInstance annotation = injectionPoint.getTarget().asField()
+                            .annotation(ON_OVERFLOW);
+
                     LOGGER.debugf("Emitter injection point '%s' detected, stream name: '%s'",
                             injectionPoint.getTargetInfo(), name);
-                    emitters.produce(new EmitterBuildItem(name));
+
+                    if (annotation != null) {
+                        emitters.produce(
+                                EmitterBuildItem.of(name,
+                                        annotation.value().toString(),
+                                        annotation.value("bufferSize").asInt()));
```

Comment:
`annotation.value("bufferSize")` may return `null` if the value is not specified. You should either use `org.jboss.jandex.AnnotationInstance.valueWithDefault(IndexView, String)` or handle `null` value, e.g. with a hardcoded default value.

Reply:
Done.